File: src/Jbuisson.NpcMaker.Angular/src/app/shared/models/character.ts
export interface ICharacter {
  Id: number;
  Name: string;
  Race: string;
  Gender: string;
  ArmorClass: string;
  HitPoints: string;
  Speed: string;
  Strength: string;
  Dexterity: string;
  Constitution: string;
  Intelligence: string;
  Wisdom: string;
  Charisma: string;
  CreatedAt: Date;

  Language: string[];

  Skills: ISkill[];
  Abilities: IAbility[];
  Properties: IProperty[];
  SavingThrows: ISavingThrow[];
};

interface IProperty {
  Name: string;
  Description: string;
};

interface IAbility {
  Name: string;
  Description: string;
};

interface ISkill {
  Name: string;
  Modifier: number;
};

interface ISavingThrow {
  Name: string;
  Modifier: number;
};

File: src/Jbuisson.NpcMaker.Data/Entities/Character.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Jbuisson.NpcMaker.Data.Entities
{
    public class Character : Entity
    {
        public string Name { get; set; }

        public string FriendlyName { get; set; }

        public string Race { get; set; }

        public bool? Gender { get; set; }

        public string Description { get; set; }

        public string Avatar { get; set; }

        public byte ArmorClass { get; set; }

        public byte HitPoints { get; set; }

        public string Speed { get; set; }

        public byte Strength { get; set; }

        public byte Dexterity { get; set; }

        public byte Constitution { get; set; }

        public byte Intelligence { get; set; }

        public byte Wisdom { get; set; }

        public byte Charisma { get; set; }

        public ICollection<Character_Skill> Skills { get; set; }

        public ICollection<Character_Language> Languages { get; set; }

        public ICollection<Character_Ability> Abilities { get; set; }

        public ICollection<Character_SavingThrow> SavingThrows { get; set; }

        public ICollection<Property> Properties { get; set; }
    }
}
